### PR TITLE
Add data.Frame event

### PR DESCRIPTION
**Pull request by jafyvilla**
The new event defines a data frame range, containing
the 'start' and 'end' frame numbers coming from a certain
data source.

### CODE REVIEW

File: zerobuf/data/frame.fbs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2016, Human Brain Project
+//                     Jafet.VillafrancaDiaz@epfl.ch
+//
+namespace zerobuf.data;
+
+// Event containing the data frame range information. It is based on the same
+// units as the zerobuf.render.Frame, i.e. number of frames.
+// It may be used as a base event that defines the available range of frames
+// coming from a certain data source, being also possible to use the
+// zerobuf.render.Frame event at the same time, which would define the frame
```
Comment (bilgili) [inline zerobuf/data/frame.fbs]: This all could have been handled in the GUI, why did we add a new event ?
The GUI is the one who commands a frame ?

Comment (jafyvilla) [inline zerobuf/data/frame.fbs]: The GUI only knows what it receives from Livre, which at the moment is the render.Frame event (start, current, end, delta). Now, Livre and the GUI are using this interval of frames to loop between them in an animation, and this range can also be modified from the GUI, being possible to focus on a specific segment of the visualization.
This new data.Frame event is adding semantics for the range of frames that the data source provides, so we can easily know the available range from the GUI, but at the same time change the range within which the rendering should animate.

**Comment by eile** [comment]
+1

File: zerobuf/data/frame.fbs (hunk 1 above)
Comment (bilgili) [inline zerobuf/data/frame.fbs]: We ( @favreau and I ) had this conclusion 6 months ago and it was rejected :) Anyways good to have it.

**Pull request merged**